src/autometrics/decorator.py

```diff
@@ -29,49 +36,103 @@ def autometrics(
     *,
     objective: Optional[Objective] = None,
 ):
+    def track_result_ok(start_time: float, function: str, module: str, caller: str):
+        get_tracker().finish(
+            start_time,
+            function=function,
+            module=module,
+            caller=caller,
+            objective=objective,
+            result=Result.OK,
+        )
+
+    def track_result_error(
+        exception: Exception,
+        start_time: float,
+        function: str,
+        module: str,
+        caller: str,
+    ):
+        get_tracker().finish(
+            start_time,
+            function=function,
+            module=module,
+            caller=caller,
+            objective=objective,
+            result=Result.ERROR,
+        )
+        # Reraise exception
+        raise exception
+
     """Decorator for tracking function calls and duration."""
 
     def decorator(func: Callable[P, T]) -> Callable[P, T]:
         module_name = get_module_name(func)
         func_name = func.__name__
 
         @wraps(func)
-        def wrapper(*args: P.args, **kwds: P.kwargs) -> T:
+        def sync_wrapper(*args: P.args, **kwds: P.kwargs) -> T:
             start_time = time.time()
             caller = get_caller_function()
 
             try:
                 result = func(*args, **kwds)
-                get_tracker().finish(
+                track_result_ok(
+                    start_time, function=func_name, module=module_name, caller=caller
+                )
+
+            except Exception as exception:
+                result = exception.__class__.__name__
+                track_result_error(
+                    exception,
                     start_time,
                     function=func_name,
                     module=module_name,
                     caller=caller,
-                    objective=objective,
-                    result=Result.OK,
+                )
+                # Reraise exception
+                raise exception
+            return result
+
+        sync_wrapper.__doc__ = append_docs_to_docstring(func, func_name, module_name)
+        return sync_wrapper
+
+    """Async Decorator for tracking function async calls and duration."""
```

Comment:
docstrings should be on the first line of the function. I think this one is added because the docstring above is confusing?

Reply:
indeed! this was the result of copy-pasting. will fix